// Copyright 2023 Intrinsic Innovation LLC
// Intrinsic Proprietary and Confidential
// Provided subject to written agreement between the parties.

syntax = "proto3";

package xfa.icon.external_actions;

// Fixed parameters for SineWaveAction.
// This action calculates the value of a sine wave with the given parameters
// each cycle, and adds the result to the starting position for each joint.
message SineWaveFixedParams {
  message JointSineParams {
    // Frequency of the sine wave for this joint.
    double frequency_hz = 1;
    // Maximum amplitude of the sine wave (applied as an offset to the position
    // the joint was in when the action started), in radians.
    double amplitude_rad = 2;
  }
  repeated JointSineParams joints = 1;
}
